lecture_1/homework/app/controllers/samurais_controller.rb

```diff
@@ -0,0 +1,67 @@
+class SamuraisController < ApplicationController
+  before_action :set_samurai, only: %i[show update destroy]
+  before_action :set_clan,    only: %i[index create active inactive]
+
+  def index
+    samurais ||= @clan.samurais.first(limit_param.to_i)
+    render json: samurais.to_json(only: %w[id name])
+  end
+
+  def show
+    render json: @samurai.to_json(except: %w[created_at updated_at])
+  end
+
+  def create
+    samurai ||= @clan.samurais.new(samurai_params)
+    if samurai.save
+      render json: samurai.to_json(except: %w[created_at updated_at]), status: 201
+    else
+      render json: { message: "Something went wrong!",
+                     status: 422,
+                     errors: samurai.errors.full_messages }, status: 422
+    end
+  end
+
+  def update
+    if @samurai.update_attributes(samurai_params)
+      render json: @samurai.to_json(except: %w[created_at updated_at])
+    else
+      render json: { message: "Something went wrong!",
+                     status: 422,
+                     errors: @samurai.errors.full_messages }, status: 422
+    end
+  end
+
+  def destroy
+    @samurai.destroy!
+    head 204
+  end
+
+  def active
```

Comment:
Ładniej by było korzystać z crud i np wrzucić to w zagnieżdżony głebiej kontroler Samurais::ActiveController#index lub jeśli chcemy uniknąć zagnieżdżenia zrealizować to w index, generalnie warto trzymać w kontrolerach API porządek, trzymać się crudowych akcji, a ładnie się potem te apki rozbudowują :D